aws/database/multi_table/patch_asset_tenures.py

```diff
@@ -0,0 +1,358 @@
+            )
+            continue
+
+        if confirm(
+            f"Update asset search tenure for property {fixup['prop_id']} from \n"
+            f"{json.dumps(current_asset.get("tenure", {}), indent=4)} -> \n"
+            f"{json.dumps(proposed_new_tenure, indent=4)}?"
+        ):
+            es_client.update(
+                index_, fixup["prop_id"], body={"doc": {"tenure": proposed_new_tenure}}
+            )
+
+
+def verify_work_complete(stage: Stage, fixups: list[Fixup]) -> None:
+    session = get_session_for_stage(stage)
+    ssm = session.client("ssm")
+    asset_api_url = get_asset_api_url(stage, ssm)
+    search_api_url = get_search_api_url(stage, ssm)
+
+    for fixup in reversed(fixups):
+        print(f"Verifying property {fixup['prop_id']}")
+        expected_tenure = fixup["new_tenures"][0]
+
+        # Check asset API
+        api_asset = get_asset_from_asset_api(asset_api_url, fixup["prop_id"])
+        assert api_asset["tenure"]["id"] == expected_tenure["id"], api_asset
+        assert expected_tenure["type"] in api_asset["tenure"]["type"], api_asset
+
+        # Check search API
+        search_asset = get_asset_from_search_api(search_api_url, fixup["prop_id"])
+        assert search_asset["tenure"]["id"] == expected_tenure["id"], search_asset
+        assert expected_tenure["type"] in search_asset["tenure"]["type"], search_asset
+
+        try:
+            assert "leasehold" not in api_asset["tenure"]["type"].lower(), api_asset
+            assert (
+                "leasehold" not in search_asset["tenure"]["type"].lower()
+            ), search_asset
+        except AssertionError:
+            print(
+                f"WARNING: Property {fixup['prop_id']} - {fixup['prop_ref']} has a leasehold"
+            )
+
+        print(f"Property {fixup['prop_id']} has been updated correctly")
+
+
+def main():
+    stage = Stage.HOUSING_DEVELOPMENT
+    session = get_session_for_stage(stage)
+
+    # Part 1: Generate fixups
+    fixups = generate_fixups(stage, session)
+    with open(os.path.join(FILE_DIR, "fixups.json"), "w") as file:
+        json.dump(fixups, file, indent=4)
+
+    if not confirm("After reviewing the fixups.json file, continue?"):
+        return
+    # You can quit the script here to take time to review the fixups.json file - it will be saved in the FIXUP_FILE_DIR
+
+    # Part 2: Update asset tenures in DynamoDB and Elasticsearch after manual fixup review
+    with open(os.path.join(FILE_DIR, "fixups.json"), "r") as file:
+        fixups = cast(list[Fixup], json.load(file))
+    update_asset_tenures_from_fixups(session, fixups)
+    if confirm(
+        f"Ensure you are port forwarding to the housing search elasticsearch on {ES_PORT}. "
+        "Update Elasticsearch with the same changes?"
+    ):
+        update_elasticsearch_from_fixups(fixups)
+
+    # Part 3: Verify that the updates have been correctly applied
+    verify_work_complete(stage, fixups)
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
Bunch of `assert` statements in this file to try and bring the script to a halt in case it is at risk of doing something unexpected